core/src/main/java/org/apache/accumulo/core/iterators/system/DeletingIterator.java

```diff
@@ -105,4 +110,30 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
       IteratorEnvironment env) {
     throw new UnsupportedOperationException();
   }
+
+  public static SortedKeyValueIterator<Key,Value> wrap(SortedKeyValueIterator<Key,Value> source,
+      boolean propogateDeletes, Behavior behavior) {
+    switch (behavior) {
+      case PROCESS:
+        return new DeletingIterator(source, propogateDeletes);
+      case FAIL:
+        return new ServerWrappingIterator(source) {
+          @Override
+          public Key getTopKey() {
+            Key top = source.getTopKey();
+            if (top.isDeleted()) {
+              throw new IllegalStateException("Saw unexpected delete " + top);
+            }
+            return top;
+          }
+        };
+      default:
+        throw new IllegalArgumentException("Unknown behavior " + behavior);
+    }
+  }
+
+  public static Behavior getBehavior(AccumuloConfiguration conf) {
```

Comment:
I would have expected this to appear on the `ConfigurationTypeHelper` class, with the other static methods that do the same kinds of thing. Putting it here makes sense in this case... but we might want to genericize this pattern of "enum" types with case-conversion and `valueOf()` lookup for generic enums there, and then this method's implementation becomes something like: `return ConfigurationTypeHelper.getEnum(Property.TABLE_DELETE_BEHAVIOR, Behavior.class);` or similar.